packages/bitcore-node/src/models/block.ts

```diff
@@ -136,17 +138,30 @@ export class Block extends BaseModel<IBlock> {
 
   async handleReorg(params: { header?: Bitcoin.Block.HeaderObj; chain: string; network: string }): Promise<boolean> {
     const { header, chain, network } = params;
-    const localTip = await this.getLocalTip(params);
+    let localTip = await this.getLocalTip(params);
     if (header && localTip && localTip.hash === header.prevHash) {
       return false;
     }
     if (!localTip || localTip.height === 0) {
       return false;
     }
-    logger.info(`Resetting tip to ${localTip.previousBlockHash}`, { chain, network });
-    await this.collection.deleteMany({ chain, network, height: { $gte: localTip.height } });
-    await TransactionModel.collection.deleteMany({ chain, network, blockHeight: { $gte: localTip.height } });
-    await CoinModel.collection.deleteMany({ chain, network, mintHeight: { $gte: localTip.height } });
+    if (header) {
```

Comment:
This if statement logic here is not needed since it throws a lot of useful blocks (why `100`?) and introduces a lot of delay to sync up at least the `100` blocks that were dropped. It highly impacts the performance. The previous logic of just dropping the tip block and relying on `p2p.sync()` works fine I think.